documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -2581,4 +2584,307 @@ The following would be valid:
+
+**How to fix the violation**:  For each operations API ,provide a schema which consistent with the above contract.
+
+The following response is a good example::
+
+```json
+...
+ "AvailableOperations": {
+    "description": "Available operations of the service",
+    "type": "object",
+    "properties": {
+      "value": {
+        "description": "Collection of available operation details",
+        "uniqueItems": false,
+        "type": "array",
+        "items": {
+          "$ref": "#/definitions/OperationDetail"
+        }
+      },
+      "nextLink": {
+        "description": "URL client should use to fetch the next page (per server side paging).\r\nIt's null for now, added for future use.",
+        "type": "string"
+      }
+    }
+  },
+  "OperationDetail": {
+    "description": "Operation detail payload",
+    "type": "object",
+    "properties": {
+      "name": {
+        "description": "Name of the operation",
+        "type": "string"
+      },
+      "isDataAction": {
+        "description": "Indicates whether the operation is a data action",
+        "type": "boolean"
+      },
+      "display": {
+        "$ref": "#/definitions/OperationDisplay",
+        "description": "Display of the operation"
+      },
+      "origin": {
+        "description": "Origin of the operation",
+        "type": "string"
+      },
+      "properties": {
+        "$ref": "#/definitions/OperationProperties",
+        "description": "Properties of the operation"
+      }
+    }
+  },
+  "OperationDisplay": {
+    "description": "Operation display payload",
+    "type": "object",
+    "properties": {
+      "provider": {
+        "description": "Resource provider of the operation",
+        "type": "string"
+      },
+      "resource": {
+        "description": "Resource of the operation",
+        "type": "string"
+      },
+      "operation": {
+        "description": "Localized friendly name for the operation",
+        "type": "string"
+      },
+      "description": {
+        "description": "Localized friendly description for the operation",
+        "type": "string"
+      }
+    }
+  }
+....
+```
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [ARM](#arm-violations): [Errors](#arm-errors) or [Warnings](#arm-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="r4019" ></a>R4019 GetCollectionResponseSchema
+
+**Category** : ARM Error
+
+**Applies to** : ARM OpenAPI(swagger) specs
+
+**Output Message** : The response in the GET collection operation "{0}" is different than the response definition in the individual GET operation "{1}".
+
+**Description** : For all resources (top-level and nested), collection GETs should have a response definition with a single property "value" containing an array of the "resource" schema.The definition returned in the collection "value" array should be the same as the response body for the individual GET.
```

Comment:
>response definition with a single property "value" [](start = 90, length = 50)

this isn't technically true, the definition should contain a "nextLink" property as well in most cases

Reply:
thanks，I  removed the word "single" to avoid the confusion